cmake/modules/FindLibElf.cmake

```diff
@@ -0,0 +1,48 @@
+# FindLibElf
+# -----------
+#
+# Find libelf library and headers
+#
+# Result Variables
+# ^^^^^^^^^^^^^^^^
+#
+# This will define the following variables:
+#
+#  LibElf_FOUND          - True if the system has libelf
+#  LibElf_INCLUDE_DIRS   - Include directories needed to use libelf
+#  LibElf_LIBRARIES      - Libraries needed to link to libelf
+#
+# and the following imported targets:
+#
+#  LibElf::LibElf        - The libelf library
+
+find_path(LibElf_INCLUDE_DIR
+  NAMES libelf.h
+  PATHS /usr/include /usr/local/include
+  PATH_SUFFIXES libelf
+)
+
+find_library(LibElf_LIBRARY
+  NAMES elf
+  PATHS /usr/lib /usr/local/lib /usr/lib/x86_64-linux-gnu
```

Comment:
i think this is the default behavior of CMake. see https://cmake.org/cmake/help/latest/command/find_library.html, and search for "the search process is as follows:".